`DiscreteModeChoiceConfigGroup`: empty-string list/set params are parsed as `{""}` instead of empty set (can crash DMC)

## Description

It looks like several comma-separated *set/list-like* parameters in `DiscreteModeChoiceConfigGroup` treat an empty XML value (`value=""`) as a set containing a **single empty string element** (`{""}`), instead of an **empty set**.

This becomes user-visible in particular for `tourFilters` / `tripFilters`: when they end up as `{""}`, the injector later tries to resolve a filter with name `""` and fails with:

```
java.lang.IllegalStateException: TourFilter '' does not exist.
  at org.matsim.contribs.discrete_mode_choice.modules.ModelModule.provideTourFilter(...)
```

I originally noticed this because `output_config.xml` produced from a run with default filter settings contains:

```xml
<module name="DiscreteModeChoice">
  <param name="tourFilters" value="" />
  <param name="tripFilters" value="" />
</module>
```

Re-using that output config as an input config makes the next run fail. The core issue seems to be the **parsing semantics of empty strings for these parameters**: an empty string should round-trip as an empty set, not `{""}`. (This appears closely related to the general “empty sets not loaded correctly” behavior discussed in #2225.

## Why this matters

* Output configs are sometimes used as a base for follow-up runs; failing on reload is a sharp edge.
* The same parsing pattern likely affects other DMC parameters that are configured as comma-separated sets, e.g. `tourConstraints`, `tripConstraints`, and `cachedModes` (and potentially any others using the same `split(",")` + `trim` pattern).

## Steps to reproduce

1. Configure / run MATSim with `DiscreteModeChoice` enabled.
2. Ensure a set-like DMC parameter is present in XML with an empty value, e.g.

   ```xml
   <param name="tourFilters" value="" />
   ```

   (This can occur naturally via `output_config.xml`.)
3. Load the config and run.

## Expected behavior

An empty XML value for a comma-separated set/list parameter should result in an **empty** collection (no entries), i.e. `Set.of()` / `Collections.emptySet()` semantics.

## Actual behavior

Empty XML values are parsed as a collection with one element `""` (after split/trim), i.e. `{""}`. 

For `tourFilters`, this can crash during DMC module provisioning:

```
com.google.inject.ProvisionException: ...
Caused by: java.lang.IllegalStateException: TourFilter '' does not exist.
  at org.matsim.contribs.discrete_mode_choice.modules.ModelModule.provideTourFilter(...)
```

## Likely code location

In `org.matsim.contribs.discrete_mode_choice.modules.config.DiscreteModeChoiceConfigGroup`, several `@StringSetter` methods parse comma-separated strings like:

* `setTourFiltersAsString(String)` / `setTripFiltersAsString(String)`
* `setTourConstraintsAsString(String)` / `setTripConstraintsAsString(String)`
* `setCachedModesAsString(String)`

They currently do essentially:

```java
Arrays.asList(value.split(",")).stream().map(String::trim).collect(Collectors.toSet())
```

So for `value == ""`, `split(",")` yields `[""]`, and after trim the empty token remains and is collected.

The crash path for filters is then in `org.matsim.contribs.discrete_mode_choice.modules.ModelModule.provideTourFilter(...)`, which throws when the filter name is unknown (and `""` is unknown). (This matches the stacktrace.)

## Proposed fix

In the relevant `@StringSetter` methods in `DiscreteModeChoiceConfigGroup`, treat blank input as empty and/or filter empty tokens:

* If `value == null || value.trim().isEmpty()` → set empty set
* Else: split, trim, **filter(token -> !token.isEmpty())**, collect

This would align DMC config parsing with the intuitive meaning of `value=""`, and would prevent `tourFilters=""` / `tripFilters=""` from becoming `{""}` and crashing.

---

If this direction is acceptable, I can provide a small PR that updates the relevant methods.

--- a/contribs/discrete_mode_choice/src/main/java/org/matsim/contribs/discrete_mode_choice/modules/config/ActivityHomeFinderConfigGroup.java
+++ b/contribs/discrete_mode_choice/src/main/java/org/matsim/contribs/discrete_mode_choice/modules/config/ActivityHomeFinderConfigGroup.java
@@ -37,7 +37,7 @@ public class ActivityHomeFinderConfigGroup extends ComponentConfigGroup {
 	@StringSetter(ACTIVITY_TYPES)
 	public void setActivityTypesAsString(String activityTypes) {
 		this.activityTypes = Arrays.asList(activityTypes.split(",")).stream().map(String::trim)
-				.collect(Collectors.toSet());
+				.filter(s -> !s.isEmpty()).collect(Collectors.toSet());
 	}
 
 	public Collection<String> getActivityTypes() {

--- a/contribs/discrete_mode_choice/src/main/java/org/matsim/contribs/discrete_mode_choice/modules/config/ActivityTourFinderConfigGroup.java
+++ b/contribs/discrete_mode_choice/src/main/java/org/matsim/contribs/discrete_mode_choice/modules/config/ActivityTourFinderConfigGroup.java
@@ -38,7 +38,7 @@ public class ActivityTourFinderConfigGroup extends ComponentConfigGroup {
 	@StringSetter(ACTIVITY_TYPES)
 	public void setActivityTypesAsString(String activityTypes) {
 		this.activityTypes = Arrays.asList(activityTypes.split(",")).stream().map(String::trim)
-				.collect(Collectors.toSet());
+				.filter(s -> !s.isEmpty()).collect(Collectors.toSet());
 	}
 
 	public Collection<String> getActivityTypes() {

--- a/contribs/discrete_mode_choice/src/main/java/org/matsim/contribs/discrete_mode_choice/modules/config/DiscreteModeChoiceConfigGroup.java
+++ b/contribs/discrete_mode_choice/src/main/java/org/matsim/contribs/discrete_mode_choice/modules/config/DiscreteModeChoiceConfigGroup.java
@@ -276,7 +276,8 @@ public class DiscreteModeChoiceConfigGroup extends ReflectiveConfigGroup {
 	 */
 	@StringSetter(TOUR_FILTERS)
 	public void setTourFiltersAsString(String tourFilters) {
-		this.tourFilters = Arrays.asList(tourFilters.split(",")).stream().map(String::trim).collect(Collectors.toSet());
+		this.tourFilters = Arrays.asList(tourFilters.split(",")).stream().map(String::trim).filter(s -> !s.isEmpty())
+				.collect(Collectors.toSet());
 	}
 
 	/**
@@ -306,7 +307,8 @@ public class DiscreteModeChoiceConfigGroup extends ReflectiveConfigGroup {
 	 */
 	@StringSetter(TRIP_FILTERS)
 	public void setTripFiltersAsString(String tripFilters) {
-		this.tripFilters = Arrays.asList(tripFilters.split(",")).stream().map(String::trim).collect(Collectors.toSet());
+		this.tripFilters = Arrays.asList(tripFilters.split(",")).stream().map(String::trim).filter(s -> !s.isEmpty())
+				.collect(Collectors.toSet());
 	}
 
 	/**
@@ -369,7 +371,7 @@ public class DiscreteModeChoiceConfigGroup extends ReflectiveConfigGroup {
 	@StringSetter(TOUR_CONSTRAINTS)
 	public void setTourConstraintsAsString(String tourConstraints) {
 		this.tourConstraints = Arrays.asList(tourConstraints.split(",")).stream().map(String::trim)
-				.collect(Collectors.toSet());
+				.filter(s -> !s.isEmpty()).collect(Collectors.toSet());
 	}
 
 	/**
@@ -400,7 +402,7 @@ public class DiscreteModeChoiceConfigGroup extends ReflectiveConfigGroup {
 	@StringSetter(TRIP_CONSTRAINTS)
 	public void setTripConstraintsAsString(String tripConstraints) {
 		this.tripConstraints = Arrays.asList(tripConstraints.split(",")).stream().map(String::trim)
-				.collect(Collectors.toSet());
+				.filter(s -> !s.isEmpty()).collect(Collectors.toSet());
 	}
 
 	/**
@@ -430,7 +432,8 @@ public class DiscreteModeChoiceConfigGroup extends ReflectiveConfigGroup {
 	 */
 	@StringSetter(CACHED_MODES)
 	public void setCachedModesAsString(String cachedModes) {
-		this.cachedModes = Arrays.asList(cachedModes.split(",")).stream().map(String::trim).collect(Collectors.toSet());
+		this.cachedModes = Arrays.asList(cachedModes.split(",")).stream().map(String::trim).filter(s -> !s.isEmpty())
+				.collect(Collectors.toSet());
 	}
 
 	/**

--- a/contribs/discrete_mode_choice/src/main/java/org/matsim/contribs/discrete_mode_choice/modules/config/LinkAttributeConstraintConfigGroup.java
+++ b/contribs/discrete_mode_choice/src/main/java/org/matsim/contribs/discrete_mode_choice/modules/config/LinkAttributeConstraintConfigGroup.java
@@ -86,7 +86,7 @@ public class LinkAttributeConstraintConfigGroup extends ComponentConfigGroup {
 	@StringSetter(CONSTRAINED_MODES)
 	public void setConstrainedModesAsString(String constrainedModes) {
 		this.constrainedModes = Arrays.asList(constrainedModes.split(",")).stream().map(String::trim)
-				.collect(Collectors.toSet());
+				.filter(s -> !s.isEmpty()).collect(Collectors.toSet());
 	}
 
 	@StringGetter(CONSTRAINED_MODES)

--- a/contribs/discrete_mode_choice/src/main/java/org/matsim/contribs/discrete_mode_choice/modules/config/MATSimTripScoringConfigGroup.java
+++ b/contribs/discrete_mode_choice/src/main/java/org/matsim/contribs/discrete_mode_choice/modules/config/MATSimTripScoringConfigGroup.java
@@ -42,7 +42,8 @@ public class MATSimTripScoringConfigGroup extends ComponentConfigGroup {
 
 	@StringSetter(PT_LEG_MODES)
 	public void setPtLegModesAsString(String ptLegModes) {
-		this.ptLegModes = Arrays.asList(ptLegModes.split(",")).stream().map(String::trim).collect(Collectors.toSet());
+		this.ptLegModes = Arrays.asList(ptLegModes.split(",")).stream().map(String::trim).filter(s -> !s.isEmpty())
+				.collect(Collectors.toSet());
 	}
 
 	@StringGetter(PT_LEG_MODES)

--- a/contribs/discrete_mode_choice/src/main/java/org/matsim/contribs/discrete_mode_choice/modules/config/ModeAvailabilityConfigGroup.java
+++ b/contribs/discrete_mode_choice/src/main/java/org/matsim/contribs/discrete_mode_choice/modules/config/ModeAvailabilityConfigGroup.java
@@ -37,7 +37,7 @@ public class ModeAvailabilityConfigGroup extends ComponentConfigGroup {
 	@StringSetter(AVAILABLE_MODES)
 	public void setAvailableModesAsString(String constrainedModes) {
 		this.availableModes = Arrays.asList(constrainedModes.split(",")).stream().map(String::trim)
-				.collect(Collectors.toSet());
+				.filter(s -> !s.isEmpty()).collect(Collectors.toSet());
 	}
 
 	@StringGetter(AVAILABLE_MODES)

--- a/contribs/discrete_mode_choice/src/main/java/org/matsim/contribs/discrete_mode_choice/modules/config/ShapeFileConstraintConfigGroup.java
+++ b/contribs/discrete_mode_choice/src/main/java/org/matsim/contribs/discrete_mode_choice/modules/config/ShapeFileConstraintConfigGroup.java
@@ -73,7 +73,7 @@ public class ShapeFileConstraintConfigGroup extends ComponentConfigGroup {
 	@StringSetter(CONSTRAINED_MODES)
 	public void setConstrainedModesAsString(String constrainedModes) {
 		this.constrainedModes = Arrays.asList(constrainedModes.split(",")).stream().map(String::trim)
-				.collect(Collectors.toSet());
+				.filter(s -> !s.isEmpty()).collect(Collectors.toSet());
 	}
 
 	@StringGetter(CONSTRAINED_MODES)

--- a/contribs/discrete_mode_choice/src/main/java/org/matsim/contribs/discrete_mode_choice/modules/config/SubtourModeConstraintConfigGroup.java
+++ b/contribs/discrete_mode_choice/src/main/java/org/matsim/contribs/discrete_mode_choice/modules/config/SubtourModeConstraintConfigGroup.java
@@ -57,7 +57,7 @@ public class SubtourModeConstraintConfigGroup extends ComponentConfigGroup {
 	@StringSetter(CONSTRAINED_MODES)
 	public void setConstrainedModesAsString(String constrainedModes) {
 		this.constrainedModes = Arrays.asList(constrainedModes.split(",")).stream().map(String::trim)
-				.collect(Collectors.toSet());
+				.filter(s -> !s.isEmpty()).collect(Collectors.toSet());
 	}
 
 	@StringGetter(CONSTRAINED_MODES)

--- a/contribs/discrete_mode_choice/src/main/java/org/matsim/contribs/discrete_mode_choice/modules/config/VehicleTourConstraintConfigGroup.java
+++ b/contribs/discrete_mode_choice/src/main/java/org/matsim/contribs/discrete_mode_choice/modules/config/VehicleTourConstraintConfigGroup.java
@@ -42,7 +42,8 @@ public class VehicleTourConstraintConfigGroup extends ComponentConfigGroup {
 	@StringSetter(RESTRICTED_MODES)
 	public void setRestrictedModesAsString(String restrictedModes) {
 		this.restrictedModes = new HashSet<>(
-				Arrays.asList(restrictedModes.split(",")).stream().map(String::trim).collect(Collectors.toSet()));
+				Arrays.asList(restrictedModes.split(",")).stream().map(String::trim).filter(s -> !s.isEmpty())
+						.collect(Collectors.toSet()));
 	}
 
 	public Collection<String> getRestrictedModes() {

--- a/contribs/discrete_mode_choice/src/main/java/org/matsim/contribs/discrete_mode_choice/modules/config/VehicleTripConstraintConfigGroup.java
+++ b/contribs/discrete_mode_choice/src/main/java/org/matsim/contribs/discrete_mode_choice/modules/config/VehicleTripConstraintConfigGroup.java
@@ -43,7 +43,8 @@ public class VehicleTripConstraintConfigGroup extends ComponentConfigGroup {
 	@StringSetter(RESTRICTED_MODES)
 	public void setRestrictedModesAsString(String restrictedModes) {
 		this.restrictedModes = new HashSet<>(
-				Arrays.asList(restrictedModes.split(",")).stream().map(String::trim).collect(Collectors.toSet()));
+				Arrays.asList(restrictedModes.split(",")).stream().map(String::trim).filter(s -> !s.isEmpty())
+						.collect(Collectors.toSet()));
 	}
 
 	public Collection<String> getRestrictedModes() {

--- a/contribs/discrete_mode_choice/src/test/java/org/matsim/contrib/discrete_mode_choice/modules/config/ConfigTest.java
+++ b/contribs/discrete_mode_choice/src/test/java/org/matsim/contrib/discrete_mode_choice/modules/config/ConfigTest.java
@@ -1,6 +1,7 @@
 package org.matsim.contrib.discrete_mode_choice.modules.config;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 
 import java.io.BufferedReader;
@@ -13,7 +14,18 @@ import java.util.HashSet;
 
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.RegisterExtension;
+import org.matsim.contribs.discrete_mode_choice.components.constraints.LinkAttributeConstraint;
+import org.matsim.contribs.discrete_mode_choice.components.constraints.ShapeFileConstraint;
+import org.matsim.contribs.discrete_mode_choice.model.DiscreteModeChoiceModel.FallbackBehaviour;
+import org.matsim.contribs.discrete_mode_choice.modules.ModelModule;
+import org.matsim.contribs.discrete_mode_choice.modules.config.ActivityTourFinderConfigGroup;
 import org.matsim.contribs.discrete_mode_choice.modules.config.DiscreteModeChoiceConfigGroup;
+import org.matsim.contribs.discrete_mode_choice.modules.config.LinkAttributeConstraintConfigGroup;
+import org.matsim.contribs.discrete_mode_choice.modules.config.MATSimTripScoringConfigGroup;
+import org.matsim.contribs.discrete_mode_choice.modules.config.MultinomialLogitSelectorConfigGroup;
+import org.matsim.contribs.discrete_mode_choice.modules.config.ShapeFileConstraintConfigGroup;
+import org.matsim.contribs.discrete_mode_choice.modules.config.VehicleTourConstraintConfigGroup;
+import org.matsim.contribs.discrete_mode_choice.modules.config.VehicleTripConstraintConfigGroup;
 import org.matsim.core.config.Config;
 import org.matsim.core.config.ConfigReader;
 import org.matsim.core.config.ConfigUtils;
@@ -23,6 +35,260 @@ import org.matsim.testcases.MatsimTestUtils;
 public class ConfigTest {
 	@RegisterExtension
 	public final MatsimTestUtils utils = new MatsimTestUtils();
+
+	@Test
+	void testDefaultValuesRoundTrip() {
+		// Create config with all default values
+		DiscreteModeChoiceConfigGroup dmcConfig = new DiscreteModeChoiceConfigGroup();
+		Config config = ConfigUtils.createConfig(dmcConfig);
+
+		// Write to file and read back
+		String path = utils.getOutputDirectory() + "/config_default.xml";
+		new ConfigWriter(config).write(path);
+
+		DiscreteModeChoiceConfigGroup dmcConfig2 = new DiscreteModeChoiceConfigGroup();
+		ConfigUtils.loadConfig(path, dmcConfig2);
+
+		// Verify top-level defaults (compare against original dmcConfig, not hard-coded values)
+		assertEquals(dmcConfig.getPerformReroute(), dmcConfig2.getPerformReroute());
+		assertEquals(dmcConfig.getEnforceSinglePlan(), dmcConfig2.getEnforceSinglePlan());
+		assertEquals(dmcConfig.getAccumulateEstimationDelays(), dmcConfig2.getAccumulateEstimationDelays());
+		assertEquals(dmcConfig.getModelType(), dmcConfig2.getModelType());
+		assertEquals(dmcConfig.getFallbackBehaviour(), dmcConfig2.getFallbackBehaviour());
+		assertEquals(dmcConfig.getModeAvailability(), dmcConfig2.getModeAvailability());
+		assertEquals(dmcConfig.getTourFinder(), dmcConfig2.getTourFinder());
+		assertEquals(dmcConfig.getHomeFinder(), dmcConfig2.getHomeFinder());
+		assertEquals(dmcConfig.getSelector(), dmcConfig2.getSelector());
+		assertEquals(new HashSet<>(dmcConfig.getTourConstraints()),
+				new HashSet<>(dmcConfig2.getTourConstraints()));
+		assertEquals(new HashSet<>(dmcConfig.getTripConstraints()),
+				new HashSet<>(dmcConfig2.getTripConstraints()));
+		assertEquals(dmcConfig.getTourEstimator(), dmcConfig2.getTourEstimator());
+		assertEquals(dmcConfig.getTripEstimator(), dmcConfig2.getTripEstimator());
+		assertEquals(new HashSet<>(dmcConfig.getTourFilters()), new HashSet<>(dmcConfig2.getTourFilters()));
+		assertEquals(new HashSet<>(dmcConfig.getTripFilters()), new HashSet<>(dmcConfig2.getTripFilters()));
+		assertEquals(new HashSet<>(dmcConfig.getCachedModes()), new HashSet<>(dmcConfig2.getCachedModes()));
+		assertEquals(dmcConfig.getWriteUtilitiesInterval(), dmcConfig2.getWriteUtilitiesInterval());
+
+		// Verify LinkAttributeConstraintConfigGroup defaults
+		LinkAttributeConstraintConfigGroup linkAttr = dmcConfig.getLinkAttributeConstraintConfigGroup();
+		LinkAttributeConstraintConfigGroup linkAttr2 = dmcConfig2.getLinkAttributeConstraintConfigGroup();
+		assertEquals(linkAttr.getRequirement(), linkAttr2.getRequirement());
+		assertEquals(linkAttr.getAttributeName(), linkAttr2.getAttributeName());
+		assertEquals(linkAttr.getAttributeValue(), linkAttr2.getAttributeValue());
+		assertEquals(new HashSet<>(linkAttr.getConstrainedModes()), new HashSet<>(linkAttr2.getConstrainedModes()));
+
+		// Verify ShapeFileConstraintConfigGroup defaults
+		ShapeFileConstraintConfigGroup shapeFile = dmcConfig.getShapeFileConstraintConfigGroup();
+		ShapeFileConstraintConfigGroup shapeFile2 = dmcConfig2.getShapeFileConstraintConfigGroup();
+		assertEquals(shapeFile.getRequirement(), shapeFile2.getRequirement());
+		assertEquals(shapeFile.getPath(), shapeFile2.getPath());
+		assertEquals(new HashSet<>(shapeFile.getConstrainedModes()), new HashSet<>(shapeFile2.getConstrainedModes()));
+
+		// Verify MultinomialLogitSelectorConfigGroup defaults
+		MultinomialLogitSelectorConfigGroup mls = dmcConfig.getMultinomialLogitSelectorConfig();
+		MultinomialLogitSelectorConfigGroup mls2 = dmcConfig2.getMultinomialLogitSelectorConfig();
+		assertEquals(mls.getMinimumUtility(), mls2.getMinimumUtility(), 1e-9);
+		assertEquals(mls.getMaximumUtility(), mls2.getMaximumUtility(), 1e-9);
+		assertEquals(mls.getConsiderMinimumUtility(), mls2.getConsiderMinimumUtility());
+
+		// Verify VehicleTourConstraintConfigGroup defaults
+		VehicleTourConstraintConfigGroup vehicleTour = dmcConfig.getVehicleTourConstraintConfig();
+		VehicleTourConstraintConfigGroup vehicleTour2 = dmcConfig2.getVehicleTourConstraintConfig();
+		assertEquals(new HashSet<>(vehicleTour.getRestrictedModes()),
+				new HashSet<>(vehicleTour2.getRestrictedModes()));
+
+		// Verify VehicleTripConstraintConfigGroup defaults
+		VehicleTripConstraintConfigGroup vehicleTrip = dmcConfig.getVehicleTripConstraintConfig();
+		VehicleTripConstraintConfigGroup vehicleTrip2 = dmcConfig2.getVehicleTripConstraintConfig();
+		assertEquals(new HashSet<>(vehicleTrip.getRestrictedModes()),
+				new HashSet<>(vehicleTrip2.getRestrictedModes()));
+		assertEquals(vehicleTrip.getIsAdvanced(), vehicleTrip2.getIsAdvanced());
+
+		// Verify SubtourModeConstraintConfigGroup defaults
+		assertEquals(new HashSet<>(dmcConfig.getSubtourConstraintConfig().getConstrainedModes()),
+				new HashSet<>(dmcConfig2.getSubtourConstraintConfig().getConstrainedModes()));
+
+		// Verify MATSimTripScoringConfigGroup defaults
+		MATSimTripScoringConfigGroup matSimScoring = dmcConfig.getMATSimTripScoringConfigGroup();
+		MATSimTripScoringConfigGroup matSimScoring2 = dmcConfig2.getMATSimTripScoringConfigGroup();
+		assertEquals(new HashSet<>(matSimScoring.getPtLegModes()), new HashSet<>(matSimScoring2.getPtLegModes()));
+
+		// Verify TourLengthFilterConfigGroup defaults
+		assertEquals(dmcConfig.getTourLengthFilterConfigGroup().getMaximumLength(),
+				dmcConfig2.getTourLengthFilterConfigGroup().getMaximumLength());
+
+		// Verify ActivityTourFinderConfigGroup defaults
+		ActivityTourFinderConfigGroup atf = dmcConfig.getActivityTourFinderConfigGroup();
+		ActivityTourFinderConfigGroup atf2 = dmcConfig2.getActivityTourFinderConfigGroup();
+		assertEquals(new HashSet<>(atf.getActivityTypes()), new HashSet<>(atf2.getActivityTypes()));
+	}
+
+	@Test
+	void testNonDefaultValuesRoundTrip() {
+		// Create config with non-default values
+		DiscreteModeChoiceConfigGroup dmcConfig = new DiscreteModeChoiceConfigGroup();
+		Config config = ConfigUtils.createConfig(dmcConfig);
+
+		// Set non-default top-level values
+		dmcConfig.setPerformReroute(false);
+		dmcConfig.setEnforceSinglePlan(true);
+		dmcConfig.setAccumulateEstimationDelays(false);
+		dmcConfig.setModelType(ModelModule.ModelType.Trip);
+		dmcConfig.setFallbackBehaviour(FallbackBehaviour.IGNORE_AGENT);
+		dmcConfig.setModeAvailability("CustomModeAvailability");
+		dmcConfig.setTourFinder("CustomTourFinder");
+		dmcConfig.setHomeFinder("CustomHomeFinder");
+		dmcConfig.setSelector("CustomSelector");
+		dmcConfig.setTourConstraints(Arrays.asList("ConstraintA", "ConstraintB"));
+		dmcConfig.setTripConstraints(Arrays.asList("ConstraintC", "ConstraintD"));
+		dmcConfig.setTourEstimator("CustomTourEstimator");
+		dmcConfig.setTripEstimator("CustomTripEstimator");
+		dmcConfig.setTourFilters(Arrays.asList("FilterA"));
+		dmcConfig.setTripFilters(Arrays.asList("FilterB"));
+		dmcConfig.setCachedModes(Arrays.asList("car", "bike"));
+		dmcConfig.setWriteUtilitiesInterval(5);
+
+		// Set non-default sub-config values
+		MultinomialLogitSelectorConfigGroup mls = dmcConfig.getMultinomialLogitSelectorConfig();
+		mls.setMinimumUtility(-500.0);
+		mls.setMaximumUtility(500.0);
+		mls.setConsiderMinimumUtility(true);
+
+		LinkAttributeConstraintConfigGroup linkAttr = dmcConfig.getLinkAttributeConstraintConfigGroup();
+		linkAttr.setRequirement(LinkAttributeConstraint.Requirement.ORIGIN);
+		linkAttr.setAttributeName("myAttribute");
+		linkAttr.setAttributeValue("myValue");
+		linkAttr.setConstrainedModes(Arrays.asList("car"));
+
+		ShapeFileConstraintConfigGroup shapeFile = dmcConfig.getShapeFileConstraintConfigGroup();
+		shapeFile.setRequirement(ShapeFileConstraint.Requirement.DESTINATION);
+		shapeFile.setPath("/path/to/shapefile.shp");
+		shapeFile.setConstrainedModes(Arrays.asList("bike", "car"));
+
+		VehicleTourConstraintConfigGroup vehicleTour = dmcConfig.getVehicleTourConstraintConfig();
+		vehicleTour.setRestrictedModes(Arrays.asList("car"));
+
+		VehicleTripConstraintConfigGroup vehicleTrip = dmcConfig.getVehicleTripConstraintConfig();
+		vehicleTrip.setRestrictedModes(Arrays.asList("car"));
+		vehicleTrip.setIsAdvanced(false);
+
+		ActivityTourFinderConfigGroup atf = dmcConfig.getActivityTourFinderConfigGroup();
+		atf.setActivityTypes(Arrays.asList("home", "work"));
+
+		dmcConfig.getMATSimTripScoringConfigGroup().setPtLegModes(Arrays.asList("pt", "bus"));
+		dmcConfig.getTourLengthFilterConfigGroup().setMaximumLength(20);
+
+		// Write to file and read back
+		String path = utils.getOutputDirectory() + "/config_nondefault.xml";
+		new ConfigWriter(config).write(path);
+
+		DiscreteModeChoiceConfigGroup dmcConfig2 = new DiscreteModeChoiceConfigGroup();
+		ConfigUtils.loadConfig(path, dmcConfig2);
+
+		// Verify all non-default top-level values
+		assertEquals(false, dmcConfig2.getPerformReroute());
+		assertEquals(true, dmcConfig2.getEnforceSinglePlan());
+		assertEquals(false, dmcConfig2.getAccumulateEstimationDelays());
+		assertEquals(ModelModule.ModelType.Trip, dmcConfig2.getModelType());
+		assertEquals(FallbackBehaviour.IGNORE_AGENT, dmcConfig2.getFallbackBehaviour());
+		assertEquals("CustomModeAvailability", dmcConfig2.getModeAvailability());
+		assertEquals("CustomTourFinder", dmcConfig2.getTourFinder());
+		assertEquals("CustomHomeFinder", dmcConfig2.getHomeFinder());
+		assertEquals("CustomSelector", dmcConfig2.getSelector());
+		assertEquals(new HashSet<>(Arrays.asList("ConstraintA", "ConstraintB")),
+				new HashSet<>(dmcConfig2.getTourConstraints()));
+		assertEquals(new HashSet<>(Arrays.asList("ConstraintC", "ConstraintD")),
+				new HashSet<>(dmcConfig2.getTripConstraints()));
+		assertEquals("CustomTourEstimator", dmcConfig2.getTourEstimator());
+		assertEquals("CustomTripEstimator", dmcConfig2.getTripEstimator());
+		assertEquals(new HashSet<>(Arrays.asList("FilterA")), new HashSet<>(dmcConfig2.getTourFilters()));
+		assertEquals(new HashSet<>(Arrays.asList("FilterB")), new HashSet<>(dmcConfig2.getTripFilters()));
+		assertEquals(new HashSet<>(Arrays.asList("car", "bike")), new HashSet<>(dmcConfig2.getCachedModes()));
+		assertEquals(5, dmcConfig2.getWriteUtilitiesInterval());
+
+		// Verify non-default sub-config values
+		MultinomialLogitSelectorConfigGroup mls2 = dmcConfig2.getMultinomialLogitSelectorConfig();
+		assertEquals(-500.0, mls2.getMinimumUtility(), 1e-9);
+		assertEquals(500.0, mls2.getMaximumUtility(), 1e-9);
+		assertEquals(true, mls2.getConsiderMinimumUtility());
+
+		LinkAttributeConstraintConfigGroup linkAttr2 = dmcConfig2.getLinkAttributeConstraintConfigGroup();
+		assertEquals(LinkAttributeConstraint.Requirement.ORIGIN, linkAttr2.getRequirement());
+		assertEquals("myAttribute", linkAttr2.getAttributeName());
+		assertEquals("myValue", linkAttr2.getAttributeValue());
+		assertEquals(new HashSet<>(Arrays.asList("car")), new HashSet<>(linkAttr2.getConstrainedModes()));
+
+		ShapeFileConstraintConfigGroup shapeFile2 = dmcConfig2.getShapeFileConstraintConfigGroup();
+		assertEquals(ShapeFileConstraint.Requirement.DESTINATION, shapeFile2.getRequirement());
+		assertEquals("/path/to/shapefile.shp", shapeFile2.getPath());
+		assertEquals(new HashSet<>(Arrays.asList("bike", "car")), new HashSet<>(shapeFile2.getConstrainedModes()));
+
+		VehicleTourConstraintConfigGroup vehicleTour2 = dmcConfig2.getVehicleTourConstraintConfig();
+		assertEquals(new HashSet<>(Arrays.asList("car")), new HashSet<>(vehicleTour2.getRestrictedModes()));
+
+		VehicleTripConstraintConfigGroup vehicleTrip2 = dmcConfig2.getVehicleTripConstraintConfig();
+		assertEquals(new HashSet<>(Arrays.asList("car")), new HashSet<>(vehicleTrip2.getRestrictedModes()));
+		assertEquals(false, vehicleTrip2.getIsAdvanced());
+
+		ActivityTourFinderConfigGroup atf2 = dmcConfig2.getActivityTourFinderConfigGroup();
+		assertEquals(new HashSet<>(Arrays.asList("home", "work")), new HashSet<>(atf2.getActivityTypes()));
+
+		assertEquals(new HashSet<>(Arrays.asList("pt", "bus")),
+				new HashSet<>(dmcConfig2.getMATSimTripScoringConfigGroup().getPtLegModes()));
+		assertEquals(20, dmcConfig2.getTourLengthFilterConfigGroup().getMaximumLength());
+	}
+
+	@Test
+	void testNullableFieldsRoundTrip() {
+		// Verify that null fields (like attributeName, attributeValue, path) remain null after round-trip
+		DiscreteModeChoiceConfigGroup dmcConfig = new DiscreteModeChoiceConfigGroup();
+		Config config = ConfigUtils.createConfig(dmcConfig);
+
+		// Explicitly set values to null to ensure null survives the round-trip
+		dmcConfig.getLinkAttributeConstraintConfigGroup().setAttributeName(null);
+		dmcConfig.getLinkAttributeConstraintConfigGroup().setAttributeValue(null);
+		dmcConfig.getShapeFileConstraintConfigGroup().setPath(null);
+
+		String path = utils.getOutputDirectory() + "/config_nullable.xml";
+		new ConfigWriter(config).write(path);
+
+		DiscreteModeChoiceConfigGroup dmcConfig2 = new DiscreteModeChoiceConfigGroup();
+		ConfigUtils.loadConfig(path, dmcConfig2);
+
+		assertNull(dmcConfig2.getLinkAttributeConstraintConfigGroup().getAttributeName());
+		assertNull(dmcConfig2.getLinkAttributeConstraintConfigGroup().getAttributeValue());
+		assertNull(dmcConfig2.getShapeFileConstraintConfigGroup().getPath());
+	}
+
+	@Test
+	void testEmptyCollectionRoundTrip() {
+		// Test that collections that are empty by default remain empty after round-trip,
+		// and that non-default collections can be explicitly emptied and remain empty.
+		DiscreteModeChoiceConfigGroup dmcConfig = new DiscreteModeChoiceConfigGroup();
+		Config config = ConfigUtils.createConfig(dmcConfig);
+
+		// Explicitly set some non-empty-default collections to empty
+		dmcConfig.getVehicleTourConstraintConfig().setRestrictedModes(Arrays.asList());
+		dmcConfig.getVehicleTripConstraintConfig().setRestrictedModes(Arrays.asList());
+		dmcConfig.getMATSimTripScoringConfigGroup().setPtLegModes(Arrays.asList());
+
+		String path = utils.getOutputDirectory() + "/config_empty.xml";
+		new ConfigWriter(config).write(path);
+
+		DiscreteModeChoiceConfigGroup dmcConfig2 = new DiscreteModeChoiceConfigGroup();
+		ConfigUtils.loadConfig(path, dmcConfig2);
+
+		// Collections that default to empty should remain empty
+		assertTrue(dmcConfig2.getLinkAttributeConstraintConfigGroup().getConstrainedModes().isEmpty());
+		assertTrue(dmcConfig2.getShapeFileConstraintConfigGroup().getConstrainedModes().isEmpty());
+		assertTrue(dmcConfig2.getSubtourConstraintConfig().getConstrainedModes().isEmpty());
+
+		// Collections explicitly set to empty should remain empty
+		assertTrue(dmcConfig2.getVehicleTourConstraintConfig().getRestrictedModes().isEmpty());
+		assertTrue(dmcConfig2.getVehicleTripConstraintConfig().getRestrictedModes().isEmpty());
+		assertTrue(dmcConfig2.getMATSimTripScoringConfigGroup().getPtLegModes().isEmpty());
+	}
 
 	@Test
 	void testReadWriteConfig() {
@@ -82,6 +348,26 @@ public class ConfigTest {
 
 		assertEquals(1, numberOfMultinomialLogit);
 		assertEquals(1, numberOfVehicleContinuity);
+	}
+
+	@Test
+	void testEmptyStringParsedAsEmptySet() {
+		DiscreteModeChoiceConfigGroup dmcConfig = new DiscreteModeChoiceConfigGroup();
+
+		dmcConfig.setTourFiltersAsString("");
+		assertTrue(dmcConfig.getTourFilters().isEmpty(), "tourFilters should be empty for empty string input");
+
+		dmcConfig.setTripFiltersAsString("");
+		assertTrue(dmcConfig.getTripFilters().isEmpty(), "tripFilters should be empty for empty string input");
+
+		dmcConfig.setTourConstraintsAsString("");
+		assertTrue(dmcConfig.getTourConstraints().isEmpty(), "tourConstraints should be empty for empty string input");
+
+		dmcConfig.setTripConstraintsAsString("");
+		assertTrue(dmcConfig.getTripConstraints().isEmpty(), "tripConstraints should be empty for empty string input");
+
+		dmcConfig.setCachedModesAsString("");
+		assertTrue(dmcConfig.getCachedModes().isEmpty(), "cachedModes should be empty for empty string input");
 	}
 
 	@Test

